grpc/src/main/java/com/linecorp/armeria/internal/common/grpc/DefaultGrpcExceptionHandlerFunction.java

```diff
@@ -0,0 +1,273 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.internal.common.grpc;
+
+import static java.util.Objects.requireNonNull;
+
+import java.io.IOException;
+import java.nio.channels.ClosedChannelException;
+
+import com.google.common.base.Strings;
+import com.google.protobuf.InvalidProtocolBufferException;
+
+import com.linecorp.armeria.client.UnprocessedRequestException;
+import com.linecorp.armeria.client.circuitbreaker.FailFastException;
+import com.linecorp.armeria.client.grpc.GrpcClientBuilder;
+import com.linecorp.armeria.common.ClosedSessionException;
+import com.linecorp.armeria.common.ContentTooLargeException;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.TimeoutException;
+import com.linecorp.armeria.common.annotation.Nullable;
+import com.linecorp.armeria.common.grpc.GrpcExceptionHandlerFunction;
+import com.linecorp.armeria.common.grpc.GrpcStatusFunction;
+import com.linecorp.armeria.common.grpc.StackTraceElementProto;
+import com.linecorp.armeria.common.grpc.StatusCauseException;
+import com.linecorp.armeria.common.grpc.ThrowableProto;
+import com.linecorp.armeria.common.grpc.protocol.ArmeriaStatusException;
+import com.linecorp.armeria.common.stream.ClosedStreamException;
+import com.linecorp.armeria.common.util.Exceptions;
+import com.linecorp.armeria.server.RequestTimeoutException;
+import com.linecorp.armeria.server.grpc.GrpcServiceBuilder;
+
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.Status.Code;
+import io.netty.handler.codec.http2.Http2Error;
+import io.netty.handler.codec.http2.Http2Exception;
+
+public final class DefaultGrpcExceptionHandlerFunction implements GrpcExceptionHandlerFunction {
+    private static final GrpcExceptionHandlerFunction INSTANCE = new DefaultGrpcExceptionHandlerFunction();
+
+    /**
+     * Returns the default {@link GrpcExceptionHandlerFunction}. This handler is also used as the final
+     * fallback when the handler customized
+     * with either {@link GrpcClientBuilder#exceptionHandler(GrpcExceptionHandlerFunction)}
+     * or {@link GrpcServiceBuilder#exceptionHandler(GrpcExceptionHandlerFunction)} returns {@code null}.
+     * For example, the following handler basically delegates all error handling to the default handler:
+     * <pre>{@code
+     * // For GrpcClient
+     * GrpcClients
+     *   .builder("http://foo.com")
+     *   .exceptionHandler((ctx, cause, metadata) -> null)
+     *   ...
+     *
+     * // For GrpcServer
+     * GrpcService
+     *   .builder()
+     *   .exceptionHandler((ctx, cause, metadata) -> null)
+     *   ...
+     * }</pre>
+     */
+    public static GrpcExceptionHandlerFunction ofDefault() {
+        return DefaultGrpcExceptionHandlerFunction.INSTANCE;
+    }
+
+    @Override
+    public @Nullable Status apply(RequestContext ctx, Throwable cause, Metadata metadata) {
```

Comment:
This method does not return null.
```suggestion
    public Status apply(RequestContext ctx, Throwable cause, Metadata metadata) {
```